test/Interop/Cxx/templates/Inputs/template-type-parameter-not-in-signature.h

```diff
@@ -0,0 +1,25 @@
+#ifndef TEST_INTEROP_CXX_TEMPLATES_INPUTS_TEMPLATE_TYPE_PARAMETER_NOT_IN_SIGNATURE_H
+#define TEST_INTEROP_CXX_TEMPLATES_INPUTS_TEMPLATE_TYPE_PARAMETER_NOT_IN_SIGNATURE_H
+
```

Comment:
These tests are great! Thanks for being thourough :)

Two suggestions: can we get a test where there are two template params and only one is used in the signature. And a test where there's a non-type-template-param (it doesn't matter how we import it, just make sure we don't crash). (Side note: at some point we'll probably want to do something similar to this patch but with non-type-template-params.)